protocols/imap/src/test/java/org/apache/james/imap/encode/ListingEncodingUtilTest.java

```diff
@@ -16,54 +16,41 @@
  * specific language governing permissions and limitations      *
  * under the License.                                           *
  ****************************************************************/
-package org.apache.james.imap.encode;
 
-import java.util.ArrayList;
-import java.util.List;
+package org.apache.james.imap.encode;
 
 import org.apache.james.imap.encode.base.ByteImapResponseWriter;
 import org.apache.james.imap.encode.base.ImapResponseComposerImpl;
 import org.apache.james.imap.message.response.ListResponse;
+import org.apache.james.mailbox.model.MailboxMetaData;
 import org.junit.Assert;
-import org.junit.Before;
 import org.junit.Test;
 
-public class ListingEncodingUtilsTest  {
+public class ListingEncodingUtilTest {
+
+    private static final String nameParameter = "LIST";
+    private static final String typeNameParameters = "A Type Name";
 
-    final String nameParameter = "LIST";
-    final String typeNameParameters = "A Type Name";
-    
-    List<String> attributesOutput;
-        
     private ByteImapResponseWriter writer = new ByteImapResponseWriter();
     private ImapResponseComposer composer = new ImapResponseComposerImpl(writer);
-    
-    @Before
-    public void setUp() throws Exception {
-        attributesOutput = new ArrayList<>();
-      
-    }
 
     @Test
     public void testShouldAddHasChildrenToAttributes() throws Exception {
-        // Setup 
-        attributesOutput.add("\\HasChildren");
-        ListResponse input = new ListResponse(false, false, false, false, true, false, nameParameter, '.');
-            
+        // Setup
+        ListResponse input = new ListResponse(MailboxMetaData.Children.HAS_CHILDREN, MailboxMetaData.Selectability.NONE, nameParameter, '.');
+
         // Exercise
         ListingEncodingUtils.encodeListingResponse(typeNameParameters, composer, input);
         Assert.assertEquals("* A Type Name (\\HasChildren) \".\" \"LIST\"\r\n", writer.getString());
     }
-    
+
     @Test
     public void testShouldAddHasNoChildrenToAttributes() throws Exception {
-        // Setup 
-        attributesOutput.add("\\HasNoChildren");
-        ListResponse input = new ListResponse(false, false, false, false, false, true, nameParameter, '.');
-            
+        // Setup
+        ListResponse input = new ListResponse(MailboxMetaData.Children.HAS_NO_CHILDREN, MailboxMetaData.Selectability.NONE, nameParameter, '.');
```

Comment:
you probably need tests renaming

Reply:
?

Reply:
testShouldAddHasNoChildrenToAttributes is not a valid test name. If you refactor the test, the least you can do is rewriting test names (it's true of other tests in this class too)

Reply:
done